plugin/src/main/java/io/jenkins/plugins/coverage/metrics/restapi/ModifiedLinesCoverageApi.java

```diff
@@ -0,0 +1,97 @@
+package io.jenkins.plugins.coverage.metrics.restapi;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import edu.hm.hafner.coverage.FileNode;
+import edu.hm.hafner.coverage.Node;
+
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+/**
+ * Remote API to list the details of modified line coverage results.
+ */
+@ExportedBean
+class ModifiedLinesCoverageApi {
+    private final List<FileWithModifiedLines> filesWithModifiedLines;
+
+    ModifiedLinesCoverageApi(final Node node) {
+        this.filesWithModifiedLines = createListOfFilesWithModifiedLines(node);
+    }
+
+    /**
+     * Finds all files with modified lines of code in a passed {@link Node} object.
+     *
+     * @param node
+     *         containing the file tree.
+     *
+     * @return a list of {@link FileWithModifiedLines} objects.
+     */
+    private static List<FileWithModifiedLines> createListOfFilesWithModifiedLines(final Node node) {
+        var filesWithModifiedLines = new ArrayList<FileWithModifiedLines>();
+
+        for (FileNode fileNode : node.filterByModifiedLines().getAllFileNodes()) {
+            var missedLines = new ArrayList<>(fileNode.getMissedLines());
+            var partiallyCoveredLines = new ArrayList<>(fileNode.getPartiallyCoveredLines().keySet());
+            var coveredLines = new ArrayList<Integer>();
+
+            for (Integer line : fileNode.getLinesWithCoverage()) {
+                if (fileNode.getMissedOfLine(line) == 0 && fileNode.getCoveredOfLine(line) > 0) {
+                    coveredLines.add(line);
+                }
+            }
+
+            var modifiedLinesBlocks = new TreeSet<ModifiedLinesBlock>();
+
+            calculateModifiedLineBlocks(coveredLines, modifiedLinesBlocks, LineCoverageType.COVERED);
+            calculateModifiedLineBlocks(missedLines, modifiedLinesBlocks, LineCoverageType.MISSED);
+            calculateModifiedLineBlocks(partiallyCoveredLines, modifiedLinesBlocks, LineCoverageType.PARTIALLY_COVERED);
```

Comment:
It is somehow untypical in Java to use a parameter as output value. Why don't you use a return value and add the result to the list?